drivers/dma/dma_sam_xdmac.c

```diff
@@ -358,11 +358,45 @@ static int sam_xdmac_initialize(const struct device *dev)
 	return 0;
 }
 
+static int sam_xdmac_get_status(const struct device *dev, uint32_t channel, struct dma_status *status){
+    const struct sam_xdmac_dev_cfg *const dev_cfg = dev->config;
+
+    Xdmac * const xdmac = dev_cfg->regs;
+
+    uint32_t config = xdmac->XDMAC_CHID[channel].XDMAC_CC;
+
+    bool initializing = (config & XDMAC_CC_INITD_Msk)? 1 : 0;
+
+    enum dma_channel_direction dir = 0;
+    if ((config & XDMAC_CC_TYPE_Msk) == 0){
+        // mem2mem
+        dir = MEMORY_TO_MEMORY;
+    } else {
+        if ((config & XDMAC_CC_DSYNC_Msk) == XDMAC_CC_DSYNC_MEM2PER){
+            // mem2per
+            dir = MEMORY_TO_PERIPHERAL;
+        } else {
+            // per2mem
+            dir = PERIPHERAL_TO_MEMORY;
+        }
+    }
+
+    uint32_t ublen = xdmac->XDMAC_CHID[channel].XDMAC_CUBC;
+
+    //
+    status->busy = (initializing) || (ublen > 0) ? 1 : 0;
+    status->dir = dir;
+    status->pending_length = ublen;
+
+    return 0;
+}
```

Comment:
Not sure why compliance checks aren't complaining, but some formatting needs to be done and some readability improvements:

```c
static int sam_xdmac_get_status(const struct device *dev, uint32_t channel,
				struct dma_status *status)
{
	const struct sam_xdmac_dev_cfg *const dev_cfg = dev->config;

	Xdmac * const xdmac = dev_cfg->regs;
	uint32_t chan_cfg = xdmac->XDMAC_CHID[channel].XDMAC_CC;
	uint32_t ublen = xdmac->XDMAC_CHID[channel].XDMAC_CUBC;

	if ((chan_cfg & XDMAC_CC_TYPE_Msk) == 0) {
		/* mem2mem */
		status->dir = MEMORY_TO_MEMORY;
	} else if ((chan_cfg & XDMAC_CC_DSYNC_Msk) == XDMAC_CC_DSYNC_MEM2PER) {
		/* mem2per */
		status->dir = MEMORY_TO_PERIPHERAL;
	} else {
		// per2mem
		status->dir = PERIPHERAL_TO_MEMORY;
	}

	status->busy = ((chan_cfg & XDMAC_CC_INITD_Msk) != 0) || (ublen > 0);
	status->pending_length = ublen;

	return 0;
}
```

Reply:
applied your changes